paneless.go

```diff
@@ -6,47 +6,49 @@ import (
 
 	"github.com/getlantern/systray"
 	"github.com/iwittenberg/paneless/icon"
+	"github.com/skratchdot/open-golang/open"
 )
 
 func main() {
 	systray.Run(onReady, nil)
 }
 
 func onReady() {
-	preferences, err := FromJSONFile("preferences.json")
-	if err != nil {
-		log.Fatal("Couldnt read from file", err)
+	var preferences WindowPreferencesList
+	if err := preferences.FromFile("preferences.json"); err != nil {
+		log.Fatalf("failed to read preferences from preferences.json: %s", err)
 	}
 
 	systray.SetIcon(icon.Data)
 	systray.SetTitle("Rearrange")
 	systray.SetTooltip("Better Rearrangement Tool")
 
-	for _, preference := range *preferences {
-		item := systray.AddMenuItem(preference.Name, "Rearrange the windows according to this preference")
-
-		go func(item *systray.MenuItem, preferences WindowPreferences) {
+	for _, p := range preferences {
+		go func() {
 			for {
-				<-item.ClickedCh
-				SetWindowPositions(preferences)
+				i := systray.AddMenuItem(p.Name, "Rearrange the windows according to this preference")
+				<-i.ClickedCh
+				p.SetPositions()
```

Comment:
VS Code provides a warning against capturing the loop variable like this.  Also, pulling the systray.AddMenuItem in to the gofunc resulted in a non-deterministic layout of menu items, especially affecting the 3 other menu items added later.  I've let p as a param to the gofunc, but pulled i outside of it and took advantage of the capture